src/modules/blockchainApplication/explore/components/BlockchainApplicationRow/BlockchainApplicationRow.js

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useMemo } from 'react';
+import grid from 'flexboxgrid/dist/flexboxgrid.css';
+import { usePinBlockchainApplication } from '@blockchainApplication/manage/hooks/usePinBlockchainApplication';
+import DialogLink from 'src/theme/dialog/link';
+import { TertiaryButton } from 'src/theme/buttons';
+import Icon from 'src/theme/Icon';
+import TokenAmount from 'src/modules/token/fungible/components/tokenAmount';
+import liskLogo from '../../../../../../setup/react/assets/images/LISK.png';
+import styles from './BlockchainApplicationRow.css';
+
+const DepositAmount = ({ amount }) => (
+  <span className={`deposit-amount ${styles.amount} ${grid['col-xs-3']}`}>
+    <TokenAmount val={amount} token="LSK" />
+  </span>
+);
+
+const ChainId = ({ id }) => (
+  <div className={`chain-id ${styles.chainId} ${grid['col-xs-3']}`}>
+    <span>{id}</span>
+  </div>
+);
+
+const ChainStatus = ({ status, t }) => (
+  <div className={grid['col-xs-2']}>
+    <span className={`chain-status ${styles.statusChip} ${styles[status]}`}>
+      {t(status)}
+    </span>
+  </div>
+);
+
+const ChainName = ({ title, logo }) => (
+  <div className={`chain-name ${grid['col-xs-4']} ${styles.chainName}`}>
+    <img src={logo} />
+    <span>{title}</span>
+  </div>
+);
+
+const Pin = ({ isPinned, onTogglePin }) => (
+  <div className={`${styles.pinWrapper} ${isPinned ? styles.show : ''}`}>
+    <TertiaryButton onClick={onTogglePin} className="blockchain-application-pin-button">
+      <Icon data-testid="pin-button" name={isPinned ? 'pinnedIcon' : 'unpinnedIcon'} />
+    </TertiaryButton>
+  </div>
+);
+
+const BlockchainApplicationRow = ({
+  data,
+  className,
+  togglePin,
+  t,
+}) => {
+  const { checkPinByChainId } = usePinBlockchainApplication();
+
+  const handleTogglePin = useCallback((event) => {
+    event.stopPropagation();
+    togglePin(data.chainID);
+  }, [togglePin]);
```

Comment:
shall we extract togglePin direct from the hooks here
```suggestion
  const {togglePin, checkPinByChainId } = usePinBlockchainApplication();

  const handleTogglePin = useCallback((event) => {
    event.stopPropagation();
    togglePin(data.chainID);
  }, [togglePin]);
```
```suggestion
  const { checkPinByChainId } = usePinBlockchainApplication();

  const handleTogglePin = useCallback((event) => {
    event.stopPropagation();
    togglePin(data.chainID);
  }, [togglePin]);
```